pkg/prometheus/promcfg_test.go

```diff
@@ -50,6 +50,144 @@ func TestConfigGeneration(t *testing.T) {
 	}
 }
 
+func TestGlobalSettings(t *testing.T) {
+	type testCase struct {
+		EvaluationInterval string
+		ScrapeInterval     string
+		HonorTimestamps    *bool
+		ExternalLabels     map[string]string
+		Expected           string
+	}
+
+	//honor := true
+	nohonor := false
+	testcases := []testCase{
+		{
+			Expected: `global:
+  evaluation_interval: 30s
+  scrape_interval: 30s
+  honor_timestamps: true
+  external_labels:
+    prometheus: /
+    prometheus_replica: $(POD_NAME)
+rule_files: []
+scrape_configs: []
+alerting:
+  alert_relabel_configs:
+  - action: labeldrop
+    regex: prometheus_replica
+  alertmanagers: []
+`,
+		},
+		{
+			EvaluationInterval: "60s",
+			Expected: `global:
+  evaluation_interval: 60s
+  scrape_interval: 30s
+  honor_timestamps: true
+  external_labels:
+    prometheus: /
+    prometheus_replica: $(POD_NAME)
+rule_files: []
+scrape_configs: []
+alerting:
+  alert_relabel_configs:
+  - action: labeldrop
+    regex: prometheus_replica
+  alertmanagers: []
+`,
+		},
+		{
+			ScrapeInterval: "60s",
+			Expected: `global:
+  evaluation_interval: 30s
+  scrape_interval: 60s
+  honor_timestamps: true
+  external_labels:
+    prometheus: /
+    prometheus_replica: $(POD_NAME)
+rule_files: []
+scrape_configs: []
+alerting:
+  alert_relabel_configs:
+  - action: labeldrop
+    regex: prometheus_replica
+  alertmanagers: []
+`,
+		},
+		{
+			HonorTimestamps: &nohonor,
+			Expected: `global:
+  evaluation_interval: 30s
+  scrape_interval: 30s
+  honor_timestamps: false
+  external_labels:
+    prometheus: /
+    prometheus_replica: $(POD_NAME)
+rule_files: []
+scrape_configs: []
+alerting:
+  alert_relabel_configs:
+  - action: labeldrop
+    regex: prometheus_replica
+  alertmanagers: []
+`,
+		},
+		{
+			ExternalLabels: map[string]string{
+				"key1": "value1",
+				"key2": "value2",
+			},
+			Expected: `global:
+  evaluation_interval: 30s
+  scrape_interval: 30s
+  honor_timestamps: true
+  external_labels:
+    key1: value1
+    key2: value2
+    prometheus: /
+    prometheus_replica: $(POD_NAME)
+rule_files: []
+scrape_configs: []
+alerting:
+  alert_relabel_configs:
+  - action: labeldrop
+    regex: prometheus_replica
+  alertmanagers: []
+`,
+		},
+	}
+
+	for _, tc := range testcases {
+		cg := &configGenerator{}
+		cfg, err := cg.generateConfig(
+			&monitoringv1.Prometheus{
+				ObjectMeta: metav1.ObjectMeta{},
+				Spec: monitoringv1.PrometheusSpec{
+					EvaluationInterval: tc.EvaluationInterval,
+					ScrapeInterval:     tc.ScrapeInterval,
+					HonorTimestamps:    tc.HonorTimestamps,
+					ExternalLabels:     tc.ExternalLabels,
+				},
+			},
+			map[string]*monitoringv1.ServiceMonitor{},
+			nil,
+			map[string]BasicAuthCredentials{},
+			map[string]BearerToken{},
+			nil,
+			nil,
+			nil,
+			nil,
+		)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if tc.Expected != string(cfg) {
+			t.Fatalf("Unexpected result.\n\nGot:\n\n%s\n\nExpected:\n\n%s\n\n", string(cfg), tc.Expected)
```

Comment:
Hmm did you mean Got instead of Unexpected?

Reply:
No, this is ok. It returns something like this when failing:
```
    promcfg_test.go:186: Unexpected result.
        
        Got:
        
        global:
          evaluation_interval: 30s
          scrape_interval: 30s
          honor_timestamps: true
          external_labels:
            key1: value1
            key2: value2
            prometheus: /
            prometheus_replica: $(POD_NAME)
        rule_files: []
        scrape_configs: []
        alerting:
          alert_relabel_configs:
          - action: labeldrop
            regex: prometheus_replica
          alertmanagers: []
        
        
        Expected:
        
        global:
          evaluation_interval: 30s
          scrape_interval: 30s
          honor_timestamps: false
          external_labels:
            key1: value1
            key2: value2
            prometheus: /
            prometheus_replica: $(POD_NAME)
        rule_files: []
        scrape_configs: []
        alerting:
          alert_relabel_configs:
          - action: labeldrop
            regex: prometheus_replica
          alertmanagers: []

```

Reply:
Yeah I noticed we seem to do this everywhere, was just confusing to read.